src/main/java/uk/gov/pay/directdebit/payments/params/PaymentViewSearchParams.java

```diff
@@ -173,7 +174,7 @@ public String generateQuery() {
     public String buildQueryParamString() {
         StringBuilder sb = new StringBuilder("");
         if (isNotBlank(mandateId)) {
-            sb.append("&" + MANDATE_ID_KEY + "=" + mandateId);
+            sb.append("&" + MANDATE_ID_EXTERNAL_KEY + "=" + mandateId);
```

Comment:
You’ve already got a `StringBuilder` here. Using `+` concatenation with a variable that isn’t a constant will create another one.